WordPress/Classes/Utility/WPUserAgent.m

```diff
@@ -35,17 +35,33 @@ - (void)setupUserAgent
     [[NSUserDefaults standardUserDefaults] registerDefaults:dictionary];
 }
 
-#pragma mark - WordPress User-Agent
+#pragma mark - Default and WordPress User-Agents
+
+- (NSString *)defaultUserAgent
+{
+    // Temporarily unset "UserAgent" from registered user defaults so that we
+    // always get the default value, independently from what's currently set as
+    // User-Agent
+    NSDictionary *originalRegisteredDefaults = [[NSUserDefaults standardUserDefaults] volatileDomainForName:NSRegistrationDomain];
+
+    NSMutableDictionary *tempRegisteredDefaults = [NSMutableDictionary dictionaryWithDictionary:originalRegisteredDefaults];
+    [tempRegisteredDefaults removeObjectForKey:WPUserAgentKeyUserAgent];
+    [[NSUserDefaults standardUserDefaults] registerDefaults:tempRegisteredDefaults];
+    
+    NSString *userAgent = [[[UIWebView alloc] init] stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+    NSAssert(userAgent != nil, @"User agent shouldn't be nil");
+    NSAssert(! [userAgent isEmpty], @"User agent shouldn't be empty");
+    
+    [[NSUserDefaults standardUserDefaults] registerDefaults:originalRegisteredDefaults];
+    
+    return userAgent;
+}
```

Comment:
@pypt - Thanks for taking the time to reply.

Regarding multi-threading support: the fact that the method is private and only used during initialization makes it OK not to be thread safe IMO.  I'm just trying to rule out obscure issues this may bring in, but it seems we're safe on this front right now.

Part of what I don't like about the code we had / have (this was not introduced by you though, so ignore this please) is that I don't think we should be using `NSUserDefaults` for this, as this is not something that needs to be persisted across app launches.  But as I said above this is an unrelated note.

Your reply on this thread is good for me.